src/main/scala/io/iohk/ethereum/db/storage/KnownNodesStorage.scala

```diff
@@ -0,0 +1,32 @@
+package io.iohk.ethereum.db.storage
+
+import java.net.URI
+
+import io.iohk.ethereum.db.dataSource.DataSource
+
+/**
+  * This class is used to store discovered nodes
+  *   Value: stored nodes list
+  */
+class KnownNodesStorage(val dataSource: DataSource) extends KeyValueStorage[String, Set[String], KnownNodesStorage]{
+  type T = KnownNodesStorage
+
+  val key = "KnownNodes"
+
+  val namespace: IndexedSeq[Byte] = Namespaces.KnownNodesNamespace
+  def keySerializer: String => IndexedSeq[Byte] = _.getBytes
+  def valueSerializer: Set[String] => IndexedSeq[Byte] = _.mkString(" ").getBytes
+  def valueDeserializer: IndexedSeq[Byte] => Set[String] = (valueBytes: IndexedSeq[Byte]) => new String(valueBytes.toArray).split(' ').toSet
+
+  protected def apply(dataSource: DataSource): KnownNodesStorage = new KnownNodesStorage(dataSource)
+
+  def getKnownNodes(): Set[URI] = synchronized {
+    get(key).getOrElse(Set.empty).filter(_.nonEmpty).map(new URI(_))
+  }
+
+  def updateKnownNodes(toAdd: Set[URI] = Set.empty, toRemove: Set[URI] = Set.empty): KnownNodesStorage = {
+    val updated = getKnownNodes() -- toRemove ++ toAdd
```

Comment:
I do not like assertions since they may produce runtime error, but we can decide what to do if node is in `toRemove` and `toAdd` at the same time